lib/icinga/downtime.cpp

```diff
@@ -397,12 +397,12 @@ void Downtime::RemoveDowntime(const String& id, bool includeChildren, bool cance
 
 	String reason;
 
-	if (expired) {
+	if (downtime->IsExpired()) {
```

Comment:
... that param doesn't necessarily indicate expiry anymore, so you have to differentiate actual expiry and GetConfigOwnerHash() mismatch in the logs. But then I'd just adjust the expiry condition as you did here and handle the GetConfigOwnerHash() mismatch in yet another else-if with a proper reason.